### PR TITLE
Adding Support for `MULTIPOINT` and some refactoring

### PR DESCRIPTION
**Pull request by JCOR11599**
Part 1 of fix for: https://github.com/dolthub/dolt/issues/3638
It looks like PRs for this are gonna get pretty big, so I'm going to split them up for easier reviewing.

Changes:
 - greatly improved code reuse for spatial types
 - refactored code to be more readable
 - some functions were  misnamed `FROMWKT`, when they should've been `FROMTEXT`
   - `ST_POINTFROMWKT` -> `ST_POINTFROMTEXT`
   - `ST_LINEFROMWKT` -> `ST_LINEFROMTEXT`
   - `ST_POLYFROMWKT` -> `ST_POLYFROMTEXT`
 - missing function aliases
   - `ST_LINESTRINGFROMTEXT`
   - `ST_LINESTRINGFROMWKB`
   - `ST_POLYGONFROMTEXT`
   - `ST_POLYGONFROMWKB`
   - `ST_GEOMETRYFROMTEXT`
   - `ST_GEOMETRYFROMWKB`
 - these functions now work with `MULTIPOINT`
   - `ST_SWAP` 
   - `ST_SRID`
   - `ST_ASWKB`
   - `ST_MULTIPOINTFROMWKB`
   - `ST_MULTIPOINTFROMTEXT`
   - `ST_MPOINTFROMWKB`
   - `ST_MPOINTFROMTEXT`
   - `ST_GEOMFROMWKB`
   - `ST_ASGEOJSON`
   - `ST_FROMGEOJSON`
   - `ST_DIMENSION`
 - fixed bug where x and y geojson values were being swapped

### CODE REVIEW

File: sql/geometry.go [hunk 1]
```diff
@@ -34,9 +34,14 @@ type GeometryType struct {
...
 // GeometryValue is the value type returned from GeometryType, which is an interface over the following types:
-// Point, Polygon, LineString.
+// Point, Polygon, LineString, MultiPoint, MultiLineString, MultiPolygon, GeometryCollection.
 type GeometryValue interface {
 	implementsGeometryValue()
+	GetSRID() uint32
+	SetSRID(srid uint32) GeometryValue
+	Serialize() []byte
+	WriteData(buf []byte)
+	Swap() GeometryValue
```
Comment (andrew-wm-arthur) [inline sql/geometry.go:44]: add doc comments here. some of these methods are unused, I assume more changes are to come

File: sql/geometry.go [hunk 2]
```diff
@@ -70,10 +76,10 @@ const (
...
 	WKBLineID
 	WKBPolyID
-	WKBMultiPointID
-	WKBMultiLineID
-	WKBMultiPolyID
-	WKBGeoCollectionID
+	WKBMPointID
+	WKBMLineID
+	WKBMPolyID
+	WKBGeomCollectionID
```
Comment (andrew-wm-arthur) [inline sql/geometry.go:82]: these are better as is, it's hard to visually differentiate between `WKBPointId` and `WKBMPointId`

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM, couple comments

File: sql/multipoint.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	lenA := len(_a.Points)
+	lenB := len(_b.Points)
+	if lenA < lenB {
+		n = lenA
+	} else {
+		n = lenB
+	}
+
+	// Compare each point until there's a difference
+	for i := 0; i < n; i++ {
```
[the remark this answers is not in the document]
Comment (JCOR11599) [reply sql/multipoint.go:78]: although the MySQL docs seem to imply that order does not matter, it seems to
```
mysql> select multipoint(point(1,2),point(3,4)) = multipoint(point(3,4),point(1,2));
+-----------------------------------------------------------------------+
| multipoint(point(1,2),point(3,4)) = multipoint(point(3,4),point(1,2)) |
+-----------------------------------------------------------------------+
|                                                                     0 |
+-----------------------------------------------------------------------+
1 row in set (0.0003 sec)
mysql> select multipoint(point(1,2),point(3,4)) = multipoint(point(1,2),point(3,4));
+-----------------------------------------------------------------------+
| multipoint(point(1,2),point(3,4)) = multipoint(point(1,2),point(3,4)) |
+-----------------------------------------------------------------------+
|                                                                     1 |
+-----------------------------------------------------------------------+
1 row in set (0.0003 sec)
```

**Pull request merged**